client/quantum_serverless/core/job.py

```diff
@@ -223,6 +226,12 @@ def __repr__(self):
 
 def save_result(result: Dict[str, Any]):
     """Saves job results."""
+
+    version = os.environ.get(GATEWAY_PROVIDER_VERSION)
+    if version is None:
+        logging.warning("Please provide `version` of gateway.")
+        return False
```

Comment:
thinking out loud, but could we default to the latest version if no version is specified? (so if they want to use the old version they have to explicitly ask for it) 

of course, plenty of reasons why that might be a bad idea...

Reply:
I thought the same too. I couldn't decide which would be the best option. I agree with your approach too so let's do that.

Reply:
Done here @psschwei [f0e7b0e](https://github.com/Qiskit-Extensions/quantum-serverless/pull/354/commits/f0e7b0e419078e14885446bc11651f08a90d136a), let me know what you think.

Reply:
Maybe I could create a new variable `GATEWAY_PROVIDER_VERSION_DEFAULT` as we have with `OT_PROGRAM_NAME` and `OT_PROGRAM_NAME_DEFAULT`. What do you think?

Reply:
@psschwei I would like to ping you here to know your opinion about the change before continue with the work 🙏

Reply:
Yeah, with the introduction of the `DEFAULT` value that has more sense. I will write that to see how it looks 👍